pkg/kubectl/cmd/set/set_volume.go

```diff
@@ -0,0 +1,960 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package set
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	kresource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	kcoreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+)
+
+const (
+	volumePrefix = "volume-"
+)
+
+var (
+	volumeLong = templates.LongDesc(`
+		Update volumes on a pod template
+
+		This command can add, update or remove volumes from containers for any object
+		that has a pod template (deployment configs, replication controllers, or pods).
+		You can list volumes in pod or any object that has a pod template. You can
+		specify a single object or multiple, and alter volumes on all containers or
+		just those that match a given name.
+
+		If you alter a volume setting on a deployment, a deployment will be
+		triggered. Changing a replication controller will not affect running pods, and
+		you cannot change a pod's volumes once it has been created.
+
+		Volume types include:
+
+		* emptydir (empty directory) *default*: A directory allocated when the pod is
+		  created on a local host, is removed when the pod is deleted and is not copied
+			across servers
+		* hostdir (host directory): A directory with specific path on any host
+		 (requires elevated privileges)
+		* persistentvolumeclaim or pvc (persistent volume claim): Link the volume
+		  directory in the container to a persistent volume claim you have allocated by
+			name - a persistent volume claim is a request to allocate storage. Note that
+			if your claim hasn't been bound, your pods will not start.
```

Comment:
this format is on purpose?

Reply:
this format was imported from original file, I will keep it  same with k8s style